### ISSUE TITLE
Question about the prediction ability about V-JEPA 2-AC checkpoints in energy_landscape_example.ipynb

### ISSUE THREAD

**Issue by liuhuxian** [facebookresearch/vjepa2#54]
I ran the `energy_landscape_example.ipynb` file and found some issues. 
In the `energy_landscape_example.ipynb` file, you plotted the heatmap of the loss calculated by uniformly sampling in a certain range of the action space. Theoretically, the heatmap should have a minimum value at the Ground truth action \((x,y,z) = (0.09,0.03,0.08)\) point. 
However, when I expanded the plotting range, I found that this was not the case.

<img width="1111" height="938" alt="Image" src="https://github.com/user-attachments/assets/a4bd988e-296b-4db2-bf5d-b01d51dc33c2" />

<img width="1078" height="937" alt="Image" src="https://github.com/user-attachments/assets/d9e96600-cc15-4c65-b049-48a677d718e1" />

<img width="1162" height="871" alt="Image" src="https://github.com/user-attachments/assets/12c5097f-4e06-4d07-a742-fbc663212b36" />

<img width="1023" height="896" alt="Image" src="https://github.com/user-attachments/assets/b5339082-f60b-492c-ba65-4a0a0788c011" />

Correct me if I'm wrong in my logic, but it seems to be displaying poor performance at sampling actions? Maybe there's a mistake in how I've run the code or in the notebook, but I'd really like some help to understand what's going on!

Additionally, I'd like to ask about this CEM algorithm. Under the default parameters, it took me 10 seconds on a 3090 computer to get the result CEM \((x,y,z) = (0.03,0.02 0.05)\), which is quite different from the Ground truth action \((x,y,z) = (0.09,0.03,0.08)\).

<img width="2201" height="205" alt="Image" src="https://github.com/user-attachments/assets/ca0ed78b-ab08-4457-bad3-7c3fd70950eb" />

 I thought it might be because the default CEM parameters led to insufficient calculations. So, I changed the CEM parameters to increase the calculation time from 10 seconds to 2 minutes, and the result I got was still CEM \((x,y,z) = (0.06,0.07,0.07)\), which is also quite different from the Ground truth action. 

<img width="2195" height="758" alt="Image" src="https://github.com/user-attachments/assets/2e4f2832-8803-4c8c-b5db-23c923a5c338" />

I further changed the CEM parameters to increase the calculation time from 10 seconds to 4 minutes, and the result I got was CEM \((x,y,z) = (0.07,0.07,0.07)\), which is still quite different from the Ground truth action.

<img width="2191" height="761" alt="Image" src="https://github.com/user-attachments/assets/7a3d37bd-aad4-4fbc-a158-5119a676d8c4" />

 I would like to ask, why does CEM seem to not be exhibiting good sampling performance? I'm really confused why the demo you've provided doesn't seem to be displaying any good performance! Thank you in advance for your help! :)

**Comment by artemZholus** [comment]
Hi @liuhuxian ,

Based on the images you provided, I don't see any issue. Here is the reason why. 

1) The first two images show the energy landscape at two different scales and they are two dimensional, only sweeping over X and Z. Since Y is excluded, the best solution the model can find is approximately ~(0.09, 0, 0.08) which is already happening in your images.
2) the last two images show the same (as I understand) plot but at even higher scale of actions. This shows how the energy landscape of the model starts to break. This is also an expected and known phenomenon since at that scale the model starts working with actions it has never seen during training. (in training the actions are at the scale of 6-7cm at most). 
3) CEM is a randomized algorithm it's not meant to produce exactly the desired delta pose. And on top of this, another reason you are having different results is that the WM is not perfect and it is giving the only approximate solution to every start and goal pair of images. To properly evaluate the quality of the outputs you need to deploy it on the robot and execute each output of CEM as an action on the robot. Then put the resulting observation as a new starting image.

**Comment by liuhuxian** [comment]
thank you for your answering! You really help me clearing up my confusion.

**Comment by quxianjiuguo** [comment]
<img width="934" height="621" alt="Image" src="https://github.com/user-attachments/assets/747fc845-a841-4f7c-add0-3e953692ce1a" />

1. Are these hyperparameters statistical?
2. Does momentum_mean refer to the mean of the displacement offset of the robotic arm actuator during the movement?
3. If so, can you give a momentum_mean and momentum_std about the three radians of the robotic arm?

**Comment by ProxyCausal** [comment]
I'm running into similar problems with the loss with the gripper

Context
<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/c87efe1a-edfd-4c7a-b06a-c1ddf47e6c78" />

Goal
<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/269678a1-cab1-429d-ac59-51deecbcdb9b" />

1-step loss from context doing 0 action on position, orientation and varying gripper action (I believe this is a delta hence -1,1. The gripper state is 0 for fully open and 1 for fully closed). While I expect loss to be highest with some value >0 this is not the case- it is to do nothing. I guess I can try small changes in position but it's in a good position already.
With CEM I able to move successfully from some position to a good position to lift the box, but the gripper will fail to close most of the time when it's in a good position- or rarely it will happen to close due to randomness but then it won't be tight enough to actually lift the box.

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/7249aa4f-8804-4a1c-a9c2-0039233f3b87" />

Wondering if I'm doing something wrong or is this your experience @artemZholus as well, or anyone else for that matter,
Thanks!

**Comment by artemZholus** [comment]
Hi @ProxyCausal , maybe try a better crop of this image. The images you shared are 640x480. while V-JEPA2 AC works with 256x256. Instead of resizing it to the target resolution, you can crop the image such that the the object and the gripper occupies a larger part of the image. Right now I think if you simply resize these images to 256x256, the pixel difference between the goal and the initial image will be too small.

**Comment by ProxyCausal** [comment]
Edited the original b/c there was a bug but the argmin changed from gripper=0.1 to 0.2 so it's a bit better with your crop recommendation! Still not as high as I was hoping (~0.60) to be ground truth. Hoping that the CEM can hone in on the trajectory in one or two steps.

**Comment by artemZholus** [comment]
I see. in my experience, just the fact that this argmin for the gripper action is of the correct sign is enough (it is positive and we want it to be positive so that the gripper will be closed). The trick that makes it work is the closed loop planning , each action brings you a bit closer to the goal and eventually you'll reach it.
